src/app/tap.rs

```diff
@@ -66,33 +66,32 @@ where
                 let log = log.clone().with_remote(remote);
 
                 if let Conditional::Some(ref tap_svc_name) = tap_svc_name {
-                    debug!("expecting Tap client name: {:?}", tap_svc_name);
+                    trace!("expecting Tap client name: {:?}", tap_svc_name);
 
                     let is_expected_name = match session.peer_identity() {
                         Conditional::Some(ref peer_name) => {
-                            debug!("found Tap client name: {:?}", peer_name);
+                            trace!("found Tap client name: {:?}", peer_name);
                             peer_name == tap_svc_name
                         }
                         Conditional::None(reason) => {
-                            debug!("did not find Tap client name: {}", reason);
+                            trace!("did not find Tap client name: {}", reason);
                             false
                         }
                     };
 
-                    if !is_expected_name {
-                        let svc = api::tap::server::TapServer::new(
-                            proxy::grpc::unauthenticated::Unauthenticated,
-                        );
-                        let spawn =
-                            spawn_tap_service(session, future::ok(svc), log).map(|()| new_service);
+                    if is_expected_name {
+                        let svc = new_service
+                            .make_service(())
+                            .map_err(|err| error!("tap MakeService error: {}", err));
+                        let spawn = spawn_tap_service(session, svc, log).map(|()| new_service);
+
                         return future::result(spawn);
                     }
                 }
 
-                let svc = new_service
-                    .make_service(())
-                    .map_err(|err| error!("tap MakeService error: {}", err));
-                let spawn = spawn_tap_service(session, svc, log).map(|()| new_service);
+                let svc =
```

Comment:
This should be unreachable, right?  `serve_tap` should never be called without a tap service name.

Reply:
It's not unreachable right now because if the names do not match, then it falls through to this branch.

I could move this up into that check and return both services inside the `if let Some(..) ...` and then I would need to add an `unreachable!`. I hesitate to do that though, because the only reason it is unreachable is from a check that takes place in the configuration setup. I don't know I want to tie the reason for an unreachable branch to a fairly unrelated piece of code in the `src/app/config.rs`.